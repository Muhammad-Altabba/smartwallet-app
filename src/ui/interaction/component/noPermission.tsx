import React, { FC } from 'react'
import {
  Platform,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Spacing } from '../../../styles'
import {
  centeredText,
  fontLight,
  fontMain,
  textSM,
  textXL,
  textXS,
} from '../../../styles/typography'

const styles = StyleSheet.create({
  notAuthorizedOverlay: {
    alignItems: 'center',
    zIndex: 1,
    justifyContent: 'flex-start',
  },
  scanText: {
    color: Colors.sandLight,
    fontSize: textXL,
    fontFamily: fontMain,
    ...centeredText,
  },
  notAuthorizedDescription: {
    color: Colors.white,
    fontSize: textSM,
    width: 260,
    marginTop: Spacing.XS,
    fontFamily: fontLight,
    lineHeight: 24,
    ...centeredText,
  },
  enableButtonWrapper: {
    marginTop: Spacing.XXL,
  },
  enableButtonText: {
    fontSize: textXS,
    fontFamily: fontMain,
    lineHeight: 20,
    color: Platform.select({
      android: Colors.pink,
      ios: Colors.nativeIosBlue,
    }),
    textDecorationLine: Platform.select({
      android: 'underline',
      ios: 'none',
    }),
    ...centeredText,
  },
})

interface Props {
  onPressEnable: () => void
}

export const NoPermissionComponent: FC<Props> = (props: Props) => (
  <React.Fragment>
    <View style={styles.notAuthorizedOverlay}>
      <Text style={styles.scanText}>{I18n.t(strings.SCAN_QR)}</Text>
      <Text style={styles.notAuthorizedDescription}>
        {I18n.t(
          strings.ENABLE_ACCESS_SO_YOU_CAN_START_TAKING_PHOTOS_AND_VIDEOS,
        )}
      </Text>
      <TouchableOpacity
        style={styles.enableButtonWrapper}
        onPress={props.onPressEnable}>
        <Text style={styles.enableButtonText}>
          {Platform.select({
            android: I18n.t(strings.ENABLE_CAMERA_ACCESS),
            ios: I18n.t(strings.REQUEST_PERMISSION),
          })}
        </Text>
      </TouchableOpacity>
    </View>
  </React.Fragment>
)
