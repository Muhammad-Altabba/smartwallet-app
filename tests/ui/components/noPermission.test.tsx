import React from 'react'
import { NoPermissionComponent } from '../../../src/ui/interaction/component/noPermission'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'

describe('No scanner permission component', () => {
  const props = {
    onPressEnable: jest.fn(),
  }

  const noPermissionComponent = shallow(<NoPermissionComponent {...props} />)

  it('matches the snapshot on initial render', () => {
    expect(noPermissionComponent).toMatchSnapshot()
  })

  it('should call the onPressEnable prop when pressed', () => {
    // @ts-ignore
    noPermissionComponent
      .find(TouchableOpacity)
      .props()
      // @ts-ignore
      .onPress()

    expect(props.onPressEnable).toHaveBeenCalledTimes(1)
  })
})
