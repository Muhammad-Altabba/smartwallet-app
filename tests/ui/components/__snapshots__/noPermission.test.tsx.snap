// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No scanner permission component matches the snapshot on initial render 1`] = `
<Fragment>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-start",
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 239, 223)",
          "fontFamily": "TT Commons",
          "fontSize": 28,
          "textAlign": "center",
        }
      }
    >
      Scan QR
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontFamily": "TTCommons-Light",
          "fontSize": 18,
          "lineHeight": 24,
          "marginTop": 8,
          "textAlign": "center",
          "width": 260,
        }
      }
    >
      Enable access so you can start taking photos and videos
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[MockFunction]}
      style={
        Object {
          "marginTop": 48,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(10, 132, 255)",
            "fontFamily": "TT Commons",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
            "textDecorationLine": "none",
          }
        }
      >
        Request permission
      </Text>
    </TouchableOpacity>
  </View>
</Fragment>
`;
